WorldBank-Transport/DRIVER
# Improve setup docs and configs based on Jean's onboarding

## Overview

It took me two days to get DRIVER fully set up; most of that time was spent getting well enough acquainted with Vagrant/Ansible/VirtualBox/Docker/Angular so that I could debug routine errors in the toolchain, but some of it was genuine issues with the app setup that could be merged back in.

This PR attempts to pull out the useful changes I made during my onboarding and merge them back upstream. 

### Checklist


### Notes

For posterity, here are a few problems I ran into that turned out to require environment changes (i.e. changes to my setup and not the app itself):

- My home directory lives in an encrypted drive on my Ubuntu 16.04 machine, and it seems that it's not possible to mount shared directories into a VM using NFS when the directories live on this kind of encrypted drive. I had to make sure that the `DRIVER` and `ashlar` repos lived outside of my encrypted drive in order to get the mount to work. The particular error I was getting was something along the lines of `mount.nfs Error: /path/to/shared/folder is not a file or a directory`.
- Provisioning the `app` and `celery` VMs failed a few times due to routine config problems (e.g. I had a port collision on 5432 that I had to resolve by setting `DRIVER_DATABASE_PORT_5432=5433`). I wasn't aware that `vagrant up` and `vagrant reload` don't automatically rerun provisioning tasks; I spent a while trying to figure out why the loopback interface was denying my requests before I realized that large chunks of the app were missing because they still needed to be provisioned after I fixed the config errors. The solution to this is to run `vagrant reload --provision` or do as the docs say and manually retry `vagrant provision <VM>` when you're debugging provisioning problems.

## Testing Instructions

Most of these changes are tiny, so I don't think they need real testing. If you want to verify my work, you could:

1. Load up the Angular apps and check to make sure that disabling HTML5 doesn't have downstream problems (shouldn't affect production anyway I don't think, since the edit is to the example config)
2. Check the styling of the schema editor forms, particularly the geography upload, to make sure that the buttons display appropriately
3. Try running a data import using a virtualenv, the `scripts/requirements.txt` file, and the updated docs in `scripts/README.md`
